File: 2015/day8.py
import AOCInit
import util
import re

if __name__ != '__main__':
    exit()

inp = util.getInput(d=8, y=2015).splitlines()

# part 1
reD = re.compile(r'\\"|\\\\')
reHex = re.compile(r'\\x[0-9a-fA-F]{2}')

def parseLine(s):
    ss = s[1:-1]
    return len(reD.findall(ss)) + 3 * len(reHex.findall(ss)) + 2

print(sum(parseLine(l) for l in inp))

# part 2
print(sum(sum(util.sub(('"', '\\'), (1, 1), s, discard=True)) + 2 for s in inp))

File: 2015/day8.rb
inp = File.read('input8').split "\n"

# part 1
puts (inp.map do |line|
  line.scan(/\\"|\\\\/).size + 3 * line.scan(/\\x[0-9a-fA-F]{2}/).size + 2
end).sum

# part 2
puts (inp.map { |line| line.scan(/\\|"/).size + 2 }).sum
